pkg/provisioner/crdprovisioner.go

```diff
@@ -154,40 +155,29 @@ func (c *CrdProvisioner) CreateVolume(
 	volumeName = azureutils.CreateValidDiskName(volumeName, true)
 	azVolumeName := strings.ToLower(volumeName)
 
-	waiter, err := c.conditionWatcher.newConditionWaiter(ctx, azVolumeType, azVolumeName, func(obj interface{}, _ bool) (bool, error) {
-		if obj == nil {
-			return false, nil
-		}
-		azVolumeInstance := obj.(*diskv1beta1.AzVolume)
-		if azVolumeInstance.Status.Detail != nil {
-			return true, nil
-		} else if azVolumeInstance.Status.Error != nil {
-			return false, util.ErrorFromAzError(azVolumeInstance.Status.Error)
-		}
-		return false, nil
-	})
-	if err != nil {
-		return nil, err
-	}
-	defer waiter.Close()
-
-	azVolumeInstance, err := lister.Get(azVolumeName)
+	ctx, workflow := azureutils.StartWorkflow(ctx, "CrdProvisioner.CreateVolume", consts.VolumeNameLabel, volumeName, consts.PvNameKey, parameters[consts.PvNameKey])
+	defer func() { workflow.Finish(err) }()
 
+	var azVolumeInstance *diskv1beta1.AzVolume
+	azVolumeInstance, err = azVLister.Get(azVolumeName)
 	if err == nil {
 		if azVolumeInstance.Status.Detail != nil {
 			// If current request has different specifications than the existing volume, return error.
 			if !isAzVolumeSpecSameAsRequestParams(azVolumeInstance, maxMountReplicaCount, capacityRange, parameters, secrets, volumeContentSource, accessibilityReq) {
-				return nil, status.Errorf(codes.AlreadyExists, "Volume with name (%s) already exists with different specifications", volumeName)
+				err = status.Errorf(codes.AlreadyExists, "Volume with name (%s) already exists with different specifications", volumeName)
```

Comment:
Do not include information in the error message that is already available in the workflow details.